core/modules/filter/views/views_handler_field_file_uri.inc

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file
+ * Definition of views_handler_field_file_uri.
+ * Variant for use in Filter module for Image Library.
+ */
+
+/**
+ * Field handler to add rendering file paths as file URLs instead of as internal file URIs.
+ */
+class views_handler_field_file_uri extends views_handler_field_file {
+  function option_definition() {
+    $options = parent::option_definition();
+    $options['file_download_path'] = array('default' => FALSE, 'bool' => TRUE);
+    $options['image_thumbnail'] = array('default' => FALSE, 'bool' => TRUE);
+    return $options;
+  }
+
+  function options_form(&$form, &$form_state) {
+    $form['file_download_path'] = array(
+      '#title' => t('Display download path instead of file storage URI'),
+      '#description' => t('This will provide the full download URL rather than the internal filestream address.'),
```

Comment:
I'm not sure "filestream address" is accurate here. Technically it is a "stream wrapper address", but I'm not sure I'd expose language like that to the user. Just an example might be better.

Reply:
I have simplified the language and hopefully a site-builder will understand, if they are competent to use the views UI.